Rewrite the README; Rename to WebAuthn; Proper Copyright (#62)
tweag/haskell-fido2: README.md

# Haskell WebAuthn Library

This library implements the
[Web Authentication Relying Party specification Level 2][spec]. The goal of Web
Authentication (WebAuthn) is to bring passwordless login/second factor
authentication to the web, logging in using a FIDO U2F Security Key, finger
print scanner and some other authenticator.

## Setting up the environment
We assume Nix is used for the development of this library. If not using Nix,
take a look at `default.nix` for the dependencies and GHC version. All
instructions below assume the use of Nix.

### Nix Caches
We highly recommend using the IOHK Binary Cache to avoid building of several
copies of GHC. Setting up the IOHK binary cache is easily done following the
[instructions][cache] provided by IOHK.

Additionally, Tweag provides a Cachix cache for the library itself and all
non-Haskell dependencies, setting up the `tweag-haskell-fido2` cache can be

> done using the [instructions][cachix] provided by Cachix.

Reviewer: You can link to https://app.cachix.org/cache/tweag-haskell-fido2 directly

### Nix Shell
The Nix shell provides all libraries and tools required to build the library,
tests, and example server. Simply call `nix-shell` to enter a shell. If the
binary caches have been configured properly, this should take little time. If
they have not been properly configured, entering the Nix shell will take
multiple hours.

All further instructions in this README assume that you are in a Nix shell.

## Developing the Library
The [Haskell Language Server (hls)][hls] and [Ormolu][ormolu] are highly recommended
for the development of this library. To configure hls we would refer you to the
instructions specific to your editor. We also recommend enabling auto-format
using Ormolu for your editor. We do, however, also provide a bash script in
`bin/autoformat.sh` that uses Ormolu to format all Haskell source files in the
repository.

> Code not formatted using Ormolu is likely to be rejected in CI.

Reviewer: ```suggestion
Code not formatted using Ormolu will be rejected in CI.
```
